### ISSUE TITLE
Dropdown Component toggles when spacebar key is pressed.

### ISSUE THREAD

**Issue by amoghsingh** [primefaces/primereact#3363]
### Describe the bug

Whenever I use the Dropdown component. Its dropdown panel toggles when spacebar key is pressed.
Currently I am using the Dropdown in my project for which I want to use spacebar key to trigger scroll action on the dropdown. 

You can check my code at https://codesandbox.io/s/stop-spacebar-from-toggling-dropdown-ono7u3?file=/src/pages/DropdownDemo.js

I have kickstarted with it but while achieving this effect what I found is there is no keyboard support for navigation in the case od dropdown. Hence, I wrote my own code so that whenver spacebar key is pressed on the dropdown, the dropdown will scroll. 

Somehow, I am able to stop the toggle when dropdown is closed initially and I press the spacebar. But once I open the dropdown and then press the spacebar to scroll, in this case the scroll happens but the dropdown gets closed immediately.

It would have been good if I could get access to some key element or attribute of this dropdown by which I can stop the toggle  behavior caused by spacebar.

Thanks.

### Reproducer

https://codesandbox.io/s/stop-spacebar-from-toggling-dropdown-ono7u3?file=/src/pages/DropdownDemo.js

### PrimeReact version

^8.4.0

### React version

18.x

### Language

ES6

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome

### Steps to reproduce the behavior

1. Click on Dropdown icon to open it.
2. Now press spacebar key, it immediately closes.
3. Again press spacebar key, it  opens.

### Expected behavior

The Dropdown should not open or close upon pressing spacebar key. The task should be accomplished only with help of mouse click on the Dropdown.

**Comment by melloware** [comment]
That violates the ARIA specification see: https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html

**Closed:**
_Space | Opens the listbox without moving focus or changing selection._

**Open:**
_Space | Sets the value to the content of the focused option in the listbox.Closes the listbox.Sets visual focus on the combobox._
